### DISCUSSION TITLE
Make address postcode required

### DISCUSSION THREAD

**Discussion by Aslam97** [lunarphp/lunar#1059] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
I dont get any error when making a payment with auto create order that provided from `authorize()`. this because i dont provide postcode in the address.

**Describe the solution you'd like**
Make postcode in address required since `ValidateCartForOrder` required postcode to be fill.

**Comment by glennjacobs** [comment]
Not all countries have postcodes. Probably needs more discussion.

**Comment by Aslam97** [comment]
> Not all countries have postcodes. Probably needs more discussion.

if thats the case. i guess by removing `postcode` from `ValidateCartForOrder` solved it?

```php
    /**
     * Return the address rules for validation.
     *
     * @return array
     */
    private function getAddressRules()
    {
        return [
            'country_id' => 'required',
            'first_name' => 'required',
            'line_one'   => 'required',
            'city'       => 'required',
            'postcode'   => 'required',
        ];
    }
```
